primitives/runtime/src/traits.rs

```diff
@@ -530,18 +530,68 @@ morph_types! {
 	} where L::Type: Ord, M: TryMorph<L::Type, Outcome = L::Type>;
 }
 
-/// Extensible conversion trait. Generic over both source and destination types.
+/// Infallible conversion trait. Generic over both source and destination types.
 pub trait Convert<A, B> {
 	/// Make conversion.
 	fn convert(a: A) -> B;
 }
 
+/// Fallible conversion trait returning a Result. Generic over both source and destination types.
+///
+/// In case an `Error` is returned, it must always be the operand which is returned.
+pub trait TryConvert<A, B> {
+	/// Attempt the conversion and return the input value in the error case.
+	fn try_convert(a: A) -> Result<B, A>;
+}
```

Comment:
```suggestion
}

impl<A, B, T: Convert<A, B>> TryConvert<A, B> for T {
    fn try_convert(a: A) -> Result<B, A> {
        Ok(T::convert(a))
    }
}
```

Reply:
I also tried this locally, and even after deleting the other implementations, it errors in the tuple impl:

```pre
 error[E0119]: conflicting implementations of trait `TryConvert<_, _>` for type `(_, _)`
     --> work/substrate/primitives/runtime/src/traits.rs:570:1
      |
  570 | / impl<A, B> TryConvert<A, B> for Tuple {
  571 | |     fn try_convert(a: A) -> Result<B, A> {
  572 | |         for_tuples!( #(
  573 | |             let a = match Tuple::try_convert(a) {
  ...   |
  579 | |     }
  580 | | }
      | |_^ conflicting implementation for `(_, _)`
  ...
  648 |   impl<A, B, T: Convert<A, B>> TryConvert<A, B> for T {
      |   --------------------------------------------------- first implementation here
      |
      = note: downstream crates may implement trait `traits::Convert<_, _>` for type `(_, _)`